src/metabase/util/encryption.clj

```diff
@@ -90,6 +94,49 @@
                      (byte-array initialization-vector)
                      {:algorithm :aes256-cbc-hmac-sha512}))))
 
+(defn encrypt-stream
+  "Wraps a plaintext input stream into an input stream that encrypts it using AES256 CBC.
+  The encryption format is slightly different for streams vs. fixed length data"
+  {:added "0.53.0"}
+  (^InputStream [^InputStream input-stream]
+   (encrypt-stream default-secret-key input-stream))
+  (^InputStream [secret-key ^InputStream input-stream]
+   (let [spec "AES/CBC/PKCS5Padding"
+         spec-header (codecs/to-bytes (format "%-32s" spec))
+         cipher (Cipher/getInstance spec)
+         iv (nonce/random-bytes 16)]
+     (.init cipher Cipher/ENCRYPT_MODE (SecretKeySpec. (bytes/slice secret-key 32 64) "AES") (IvParameterSpec. iv))
+     (SequenceInputStream. (ByteArrayInputStream. (bytes/concat spec-header iv)) (CipherInputStream. input-stream cipher)))))
+
+(defn encrypt-for-stream
+  "Encrypts a byte-array in a way that can be used to read it with decrypt-stream instead of decrypt."
+  {:added "0.53.0"}
+  (^bytes [^bytes input]
+   (encrypt-for-stream default-secret-key input))
+  (^bytes [secret-key ^bytes input]
+   (with-open [encrypted (encrypt-stream secret-key (ByteArrayInputStream. input))]
+     (.readAllBytes encrypted))))
+
+(defn maybe-decrypt-stream
+  "Wraps a possibly-encrypted input stream into a new input stream that decrypts it if necessary."
+  {:added "0.53.0"}
+  (^InputStream [^InputStream input-stream]
+   (maybe-decrypt-stream default-secret-key input-stream))
+  (^InputStream [secret-key ^InputStream input-stream]
+   (let [spec-array (byte-array 32)
+         spec-array-length (.read input-stream spec-array)
+         spec (str/trim (codecs/bytes->str spec-array))]
+     (cond (= spec-array-length -1) input-stream
+           (and (= spec-array-length 32) (= spec "AES/CBC/PKCS5Padding"))
+           (let [cipher (Cipher/getInstance spec)
+                 iv (byte-array 16)
+                 _ (.read input-stream iv)]
+             (.init cipher Cipher/DECRYPT_MODE (SecretKeySpec. (bytes/slice secret-key 32 64) "AES") (IvParameterSpec. iv))
+             (CipherInputStream. input-stream cipher))
+           :else (SequenceInputStream.
+                  (ByteArrayInputStream. (bytes/slice spec-array 0 spec-array-length))
+                  input-stream)))))
```

Comment:
When the forms in a `cond` are long like this, it's a lot more readable to group them pairwise with empty lines (see the [style guide](https://guide.clojure.style/#short-forms-in-cond))
```suggestion
     (cond 
       (= spec-array-length -1) 
       input-stream

       (and (= spec-array-length 32) (= spec "AES/CBC/PKCS5Padding"))
       (let [cipher (Cipher/getInstance spec)
             iv (byte-array 16)
             _ (.read input-stream iv)]
         (.init cipher Cipher/DECRYPT_MODE (SecretKeySpec. (bytes/slice secret-key 32 64) "AES") (IvParameterSpec. iv))
         (CipherInputStream. input-stream cipher))

       :else 
       (SequenceInputStream.
        (ByteArrayInputStream. (bytes/slice spec-array 0 spec-array-length))
        input-stream)))))
```